registry/registry.go

```diff
@@ -388,7 +391,15 @@ func (rr *Registry) FindRegistration(c context.Context, key string) (*Registrati
 // Register under the given key. A registration can happen only
 // once, and registering more than once will return an error.
 // Hence, registration can be used for mutual-exclusion.
-func (rr *Registry) Register(c context.Context, key string, options ...Option) error {
+func (rr *Registry) Register(c context.Context, key string, annotations ...map[string]string) error {
+	if len(annotations) > 1 {
+		return ErrInvalidAnnotationsOption
+	}
+	var atns map[string]string
```

Comment:
Could we change this back into a map that is always not nil. This way the json that is inside etcd will have a {} as the value rather than nil.